### PR TITLE
Reimplemented re2post function

### PR DESCRIPTION
**Pull request by Bzdeco**

### CODE REVIEW

File: regex2tree.erl [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+-include_lib("eunit/include/eunit.hrl").
+
+%% API
+-export([re2post/1,
+         extract_from_parentheses/1]).
+
+re2post([]) -> [];
+re2post(Input) -> re2post(Input, 0, [], []).
+
+re2post([], WrittenCount, OperatorStack, Output) ->
```
Comment (OatmealLick) [inline regex2tree.erl]: is it operator tho? Aren't these for character literals?

File: regex2tree.erl [hunk 2]
```diff
@@ -0,0 +1,76 @@
+-module(regex2tree).
```
Comment (OatmealLick) [inline regex2tree.erl]: if this gonna stay change this to sth this actually does

File: regex2tree.erl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  end.
+
+is_unary_operator(Char) -> lists:member(Char, [$*]).
+
+concatenation(Output) -> [$. | Output].
+
+
+re2post_test() ->
+  [
+    ?assert(re2post("") =:= []),
```
Comment (OatmealLick) [inline regex2tree.erl]: you should use `?_assert` macro, cause it's equivalent to `?_test(?assert(...))`, ?assert does not create test object itself.

**Review by OatmealLick — CHANGES_REQUESTED** [review]

**Review by OatmealLick — APPROVED** [review]

File: regex2tree.erl (hunk 1 above)
Comment (Bzdeco) [reply regex2tree.erl]: It is indeed only for Operators, more precisely only binary operators are put there (which only the `+` for now)

File: regex2tree.erl (hunk 3 above)
Comment (Bzdeco) [reply regex2tree.erl]: Didn't know how it worked, just copied the pattern from your work `:^)`

Comment (OatmealLick) [reply regex2tree.erl]: well brainless copying didn't work HUH? 🗡

**Pull request merged**